api_app/analyzers_manager/file_analyzers/doc_info.py

```diff
@@ -172,25 +209,125 @@ def analyze_for_follina_cve(self) -> List[str]:
-            clsid = ole.getclsid(entry)
-            if clsid_text := KNOWN_CLSIDS.get(clsid.upper(), None):
-                if "cve" in clsid_text.lower():
-                    results.append(
+        try:
+            olefile.isOleFile(self.filepath)
+            ole = olefile.OleFileIO(self.filepath)
+        except olefile.olefile.NotOleFileError:
+            # not an OLE2 structured storage file
+            # do not proceed
+            return results
+        else:
+            for entry in sorted(ole.listdir(storages=True)):
+                clsid = ole.getclsid(entry)
+                if clsid_text := KNOWN_CLSIDS.get(clsid.upper(), None):
+                    if "cve" in clsid_text.lower():
+                        results.append(
+                            {
+                                "clsid": clsid,
+                                "info": clsid_text,
+                                "CVEs": list(re.findall(pattern, clsid_text)),
+                            }
+                        )
+        return results
+
+    def get_external_relationships(self) -> List:
+        external_relationships = []
+        try:
+            olefile.isOleFile(self.filepath)
+            oid = oleid.OleID(self.filepath)
+        except olefile.olefile.NotOleFileError:
+            # not an OLE2 structured storage file
+            # do not proceed
+            return external_relationships
+        else:
+            if sum(i.value for i in oid.check() if i.id == "ext_rels") > 1:
+                xml_parser = XmlParser(self.filepath)
+                for relationship, target in oleobj.find_external_relationships(
+                    xml_parser
+                ):
+                    external_relationships.append(
                         {
-                            "clsid": clsid,
-                            "info": clsid_text,
-                            "CVEs": list(re.findall(pattern, clsid_text)),
+                            "relationship": relationship,
+                            "target": target,
                         }
                     )
-        return results
+        return external_relationships
+
+    def get_docx_urls(self) -> List:
+        urls = []
+        pages_count = 0
+
+        try:
+            document = zipfile.ZipFile(self.filepath)  # check if docx document
+        except zipfile.BadZipFile as e:
+            error_message = f"job_id {self.job_id} docx bad zip file: {e}"
+            logger.warning(error_message, stack_info=True)
+            self.report.errors.append(error_message)
+        else:
+            try:
+                dxml = document.read("docProps/app.xml")
+                pages_count = int(
+                    parseString(dxml)
+                    .getElementsByTagName("Pages")[0]
+                    .childNodes[0]
+                    .nodeValue
+                )
+            except KeyError:
+                # number of pages not found, maybe the file is malformed
```

Comment:
add a log and a warning about that otherwise it's impossible to track or debug this issue at a later time